designer/client/components/FormDetails/FormDetails.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { Component, ChangeEvent, ContextType, FormEvent } from "react";
+import { Data, clone, FormConfiguration } from "@xgovformbuilder/model";
+import isFunction from "lodash/isFunction";
+
+import { validateTitle, hasValidationErrors } from "../../validations";
+import * as formConfigurationApi from "../../load-form-configurations";
+import ErrorSummary from "../../error-summary";
+import { DataContext } from "../../context";
+import { i18n } from "../../i18n";
+
+import { FormDetailsTitle } from "./FormDetailsTitle";
+import { FormDetailsFeedback } from "./FormDetailsFeedback";
+import { FormDetailsPhaseBanner } from "./FormDetailsPhaseBanner";
+import "./FormDetails.scss";
+
+type PhaseBanner = Exclude<Data["phaseBanner"], undefined>;
+type Phase = PhaseBanner["phase"];
+
+type Props = {
+  onCreate?: (saved: boolean) => void;
+};
+
+type State = {
```

Comment:
Out of curiosity why did you create a type here instead of creating an interface?

Reply:
I only use interfaces when merge declarations are needed, otherwise I use types.
https://www.typescriptlang.org/docs/handbook/advanced-types.html#interfaces-vs-type-aliases

Reply:
Fair enough. Ha, it looks like in the documentation you shared they recommend interfaces over types:
```
Because an interface more closely maps how JavaScript objects work by being open to extension, we recommend using an interface over a type alias when possible.
```

But it's not that big of a deal IMO, I was just interested.

Reply:
Right, I had missed that. Let me refactor then.